sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -2057,16 +2063,15 @@ def get_block_list(self, block_list_type="committed", **kwargs):
                 **kwargs)
         except HttpResponseError as error:
             process_storage_error(error)
-        return _get_block_list_result(blocks)
+        return _get_block_list_result(blocks)  # type: ignore
```

Comment:
Should be able to properly type `_get_block_list_result` to fix this.

Reply:
My current typing:
```
def _get_block_list_result(blocks: BlockList) -> Tuple[List[Optional[BlobBlock]], List[Optional[BlobBlock]]]:
    """Gets the block list results from the provided blocks.

    :param BlockList blocks:
        The blocks returned from generated code.
    :returns: A list of blocks.
    :rtype: Tuple[List[Optional[BlobBlock]], List[Optional[BlobBlock]]]
    """
    committed = []
    uncommitted = []
    if blocks.committed_blocks:
        committed = [BlobBlock._from_generated(b) for b in blocks.committed_blocks]  # pylint: disable=protected-access
    if blocks.uncommitted_blocks:
        uncommitted = [BlobBlock._from_generated(b) for b in blocks.uncommitted_blocks]  # pylint: disable=protected-access
    return committed, uncommitted
```

Same issue here -- how do we express that if provided -> will not be `None`

Reply:
Just drop `Optional` around `BlockBlob` in return type.

Reply:
Turns out we actually have a test for no blocks, so the original typehint here was wrong 😄